pkg/gatherers/conditional/types.go

```diff
@@ -0,0 +1,38 @@
+package conditional
+
+import "github.com/openshift/insights-operator/pkg/gatherers"
+
+// conditionType defines conditions to check
+type conditionType string
+
+// alertIsFiring is a condition to check that alert is firing.
+// Params:
+//   - name - name of the alert
+const alertIsFiring conditionType = "alert_is_firing"
+
+// conditionWithParams is a type holding a condition with its params
+type conditionWithParams struct {
+	Type   conditionType          `json:"type"`
+	Params map[string]interface{} `json:"params"`
+}
+
+// GatheringFunctionParams is a type to store gathering function parameters
+type GatheringFunctionParams map[string]interface{}
+
+// gatheringFunctionName defines functions of conditional gatherer
+type gatheringFunctionName string
```

Comment:
It looks like this type could be used more, but it would probably require some more changes. Again not a blocker.

Reply:
What do you mean used more?

Reply:
E.g https://github.com/openshift/insights-operator/pull/433/files#diff-57e9d48447e382f53bafe213301e653fd5989c354c9d751e7f6386be25459e89R251 could return `map[gatheringFunctionName]gatherers.GatheringClosure, []error` and there are similar places...but it's just a detail....